worker/uniter/resolver.go

```diff
@@ -189,6 +200,22 @@ func (s *uniterResolver) nextOpConflicted(
 	return nil, resolver.ErrWaiting
 }
 
+func (s *uniterResolver) newUpgradeOperation(
+	localState resolver.LocalState,
+	remoteState remotestate.Snapshot,
+	opFactory operation.Factory,
+) (operation.Operation, error) {
+	// Verify the charm profile before proceeding.  No hooks to run, if the
+	// correct one is not yet applied.
+	_, err := s.config.VerifyCharmProfile.NextOp(localState, remoteState, opFactory)
```

Comment:
nil check on VerifyCharmProfile or check for CAAS model

Reply:
The real thing to do is pass in the config a `CAAS` version of the `VerifyCharmProfile`, then we can get rid of the CAAS model checks from within the codebase that should be instantiated at creation time.

Reply:
verifycharmprofile.NewResolver() will now return an IAAS or a CAAS version of the resolver.

Reply:
added a check for the install code path also.